src/statisticalmodel.jl

```diff
@@ -314,3 +314,13 @@ function adjr2(model::StatisticalModel, variant::Symbol)
 end
 
 const adjr² = adjr2
+
+"""
+    momentmatrix(model::StatisticalModel)
+    
+Return the matrix containing the estimating equation.
```

Comment:
I don't understand this sentence, that can mean that this is jargon not shared across subfields. (Not an issue with the function name `momentmatrix`, just with the docstring.)

Reply:
It is a little bit tricky naming and documenting this function. For likelihood-based models, `momentmatrix` returns the matrix of log-scores, i.e. the gradient of the loglikelihood for each observation. However, for semiparametric models, the momentmatrix returns the moment conditions (for each observation) which identify the parameter vector. For instance, the IV estimator of the parameter of a linear model is `betahat=(Z'X)^(-1)Z'Y`. The `momentmatrix` returns `Z'(Y-X*betahat)`. 
This matrix is then what we need to construct a robust (to heteroskedasticity and serial correlation) variance estimator of parameters. In CovarianceMatrices.jl I have to do lots of gymnastics to get it and define an API that other packages may hook up to. 

I will try to fix the docstring to be more clear about it.

Reply:
Would this work?

```
"""
    momentmatrix(model::StatisticalModel)
    
Return the matrix containing the empirical moments defining the estimated parameters.

For likelihood-based models, `momentmatrix` returns the log-score, i.e., the log-likelihood 
gradient evaluated at each observation. For semiparametric models, `momentmatrix` 
returns a matrix whose sum along `dims=1` gives the set of equations that identify the parameter.

For linear and generalized linear models, the parameters of interest are 
the coefficients of the linear predictor. The moment matrix of a linear model is given by `u*X`, 
where `u` is the vector of residuals and `X` is the model matrix. The moment matrix of a generalized 
linear model with link function `g` is given by `X'e`, where `e` is `Y-g^(-1)(X'beta)`.
"""
```

Reply:
Maybe say "whose sum over columns"? Also though I'm not familiar with this, saying that a sum "gives the set of equations" sounds like a bit sloppy mathematically?

In the last equation, `Y` and `beta` should be defined.

Reply:
If I could write math in docstring I would write something like this:
 
```
    momentmatrix(model::StatisticalModel)
    
Return the matrix containing the empirical moments defining the estimated parameters.

For likelihood-based models, `momentmatrix` returns the log-score, i.e. the gradient
of the loglikelihood evaluated at each observation. For semiparametric models, each row 
of the `n\\times k` matrix returned by `momentmatrix` is `m(X_i, \\hat{\\beta})`, where `m(\\cdot, \\cdot)` 
is a vector valued function evaluated at the estimated parameter `\\hat{\\beta}` satisfying 
``\\sum_{i=1}^n m(x_i, \\hat{\\beta}) = 0``.

For linear and generalized linear models, the parameters of interest are 
the coefficients of the linear predictor. The moment matrix of a linear model is given by `u*X`, 
where `u` is the vector of residuals and `X` is the model matrix. The moment matrix of a generalized 
linear model with link function `g`, the moment matrix is `X'e`, where `e` is given by `Y-g^(-1)(X'beta)`, 
where `X` is the model matrix and `Y` is the model response.
```

Reply:
Fine with me. Though for math equations I think it's better to use LaTeX syntax within double backticks. The sign after β̂ isn't rendered correctly on GitHub nor at the REPL. The sum would also look better I think.

Also, is `m(⋅, ⋅)` standard notation? I'm not used to it.

Reply:
Fixed in new commit.